taichi/rhi/llvm/llvm_device.h

```diff
@@ -20,6 +20,15 @@ class LlvmDevice : public Device {
     TI_NOT_IMPLEMENTED
   }
 
+  template <typename DEVICE>
+  DEVICE *as() {
+    return dynamic_cast<DEVICE *>(this);
```

Comment:
In `class Type`, `as` asserts the dynamic cast is successful while `cast` just returns the result of the dynamic cast. We should assert here to be consistent with `Type`.